tests/parametric/test_config_consistency.py

```diff
@@ -281,30 +281,36 @@ def test_setting_trace_rate_limit(self, library_env, test_agent, test_library):
         ), "Expected at least one trace to be rate-limited with sampling priority -1."
 
 
-def tag_scenarios():
-    env1: dict = {"DD_TAGS": "key1:value1,key2:value2"}
-    env2: dict = {"DD_TAGS": "key1:value1 key2:value2"}
-    env3: dict = {"DD_TAGS": "env:test aKey:aVal bKey:bVal cKey:"}
-    env4: dict = {"DD_TAGS": "env:test,aKey:aVal,bKey:bVal,cKey:"}
-    env5: dict = {"DD_TAGS": "env:test,aKey:aVal bKey:bVal cKey:"}
-    env6: dict = {"DD_TAGS": "env:test     bKey :bVal dKey: dVal cKey:"}
-    env7: dict = {"DD_TAGS": "env :test, aKey : aVal bKey:bVal cKey:"}
-    env8: dict = {"DD_TAGS": "env:keyWithA:Semicolon bKey:bVal cKey"}
-    env9: dict = {"DD_TAGS": "env:keyWith:  , ,   Lots:Of:Semicolons "}
-    env10: dict = {"DD_TAGS": "a:b,c,d"}
-    env11: dict = {"DD_TAGS": "a,1"}
-    env12: dict = {"DD_TAGS": "a:b:c:d"}
-    return parametrize("library_env", [env1, env2, env3, env4, env5, env6, env7, env8, env9, env10, env11, env12])
+tag_scenarios: dict = {
+    "key1:value1,key2:value2": [("key1", "value1"), ("key2", "value2")],
+    "key1:value1 key2:value2": [("key1", "value1"), ("key2", "value2")],
+    "env:test aKey:aVal bKey:bVal cKey:": [("env", "test"), ("aKey", "aVal"), ("bKey", "bVal"), ("cKey", "")],
+    "env:test,aKey:aVal,bKey:bVal,cKey:": [("env", "test"), ("aKey", "aVal"), ("bKey", "bVal"), ("cKey", "")],
+    "env:test,aKey:aVal bKey:bVal cKey:": [("env", "test"), ("aKey", "aVal bKey:bVal cKey:")],
```

Comment:
Should we also add some test scenarios for keys/values as numbers, booleans, special characters, and other edge cases?

```suggestion
    "env:test,aKey:aVal bKey:bVal cKey:": [("env", "test"), ("aKey", "aVal bKey:bVal cKey:")],
    "numKey:1,1:numVal,boolKey:true,true:boolVal": [("numKey", "1"), ("1", "numVal"), ("boolKey", "true"), ("true", "boolVal") ],
    "": [],
    "sp%key:value,key:sp#value": [], # I'm not sure what belongs here
```

Reply:
Yes, but I would prefer to do this in a separate PR. Some of the existing test cases are confusing to me, so I want to revisit the existing test cases and add to them as we build consensus

Reply:
SGTM